src/Cpp/1-getting-started/14-HelloCOM/HelloCOM.cpp

```diff
@@ -0,0 +1,330 @@
+    _debugInfoQueue->SetBreakOnSeverity(D3D11_MESSAGE_SEVERITY_ERROR, true);
+
+    D3D11_MESSAGE_ID hide[] =
+    {
+        D3D11_MESSAGE_ID_SETPRIVATEDATA_CHANGINGPARAMS,
+        // Add more message IDs here as needed
+    };
+
+    D3D11_INFO_QUEUE_FILTER filter = {};
+    filter.DenyList.NumIDs = _countof(hide);
+    filter.DenyList.pIDList = hide;
+    _debugInfoQueue->AddStorageFilterEntries(&filter);
+    _debugInfoQueue.Reset();
+#endif
+
+    DXGI_SWAP_CHAIN_DESC swapchainInfo;
+    swapchainInfo.BufferDesc.Width = _width;
+    swapchainInfo.BufferDesc.Height = _height;
+    swapchainInfo.BufferDesc.RefreshRate.Numerator = 0;
+    swapchainInfo.BufferDesc.RefreshRate.Denominator = 1;
+    swapchainInfo.BufferDesc.Format = DXGI_FORMAT_B8G8R8A8_UNORM_SRGB;
+    swapchainInfo.BufferDesc.ScanlineOrdering = DXGI_MODE_SCANLINE_ORDER_UNSPECIFIED;
+    swapchainInfo.BufferDesc.Scaling = DXGI_MODE_SCALING_STRETCHED;
+    swapchainInfo.SampleDesc.Count = 1;
+    swapchainInfo.SampleDesc.Quality = 0;
+    swapchainInfo.BufferUsage = DXGI_USAGE_RENDER_TARGET_OUTPUT;
+    swapchainInfo.BufferCount = 2;
+    swapchainInfo.OutputWindow = nativeWindow;
+    swapchainInfo.Windowed = true;
+    swapchainInfo.SwapEffect = DXGI_SWAP_EFFECT_DISCARD;
+    swapchainInfo.Flags = {};
+    
+    if (FAILED(_dxgiFactory->CreateSwapChain(_device.Get(), &swapchainInfo, &_swapChain)))
+    {
+        std::cout << "DXGI: Failed to create SwapChain\n";
+        return false;
+    }
+
+    ComPtr<ID3D11Texture2D> backBuffer = nullptr;
+    if (FAILED(_swapChain->GetBuffer(0, __uuidof(ID3D11Texture2D), &backBuffer)))
+    {
+        std::cout << "D3D11: Failed to get Back Buffer from the SwapChain\n";
+        return false;
+    }
+
+    if (FAILED(_device->CreateRenderTargetView(backBuffer.Get(), nullptr, &_renderTarget)))
+    {
+        std::cout << "D3D11: Failed to create RTV from Back Buffer\n";
+        return false;
+    }
+    // This section reads and compiles both the vertex and pixel shaders
+    UINT compileFlags = D3DCOMPILE_ENABLE_STRICTNESS;
+#if !defined(NDEBUG)
+    compileFlags |= D3DCOMPILE_DEBUG;
+#endif
+    ComPtr<ID3D10Blob> vertexShaderBlob = nullptr;
+    ComPtr<ID3D10Blob> pixelShaderBlob = nullptr;
+    ComPtr<ID3D10Blob> errorBlob = nullptr;
+    if (FAILED(D3DCompileFromFile(
+        L"Assets/Shaders/Main.vs.hlsl",
+        nullptr,
+        D3D_COMPILE_STANDARD_FILE_INCLUDE,
+        "main",
+        "vs_5_0",
+        compileFlags,
+        0,
+        &vertexShaderBlob,
+        &errorBlob)))
+    {
+        std::cout << "D3D11: Failed to read vertex shader from file\n";
+        if (errorBlob)
+        {
+            std::cout << "D3D11: With message: " << static_cast<const char*>(errorBlob->GetBufferPointer()) << "\n";
+        }
+        return false;
+    }
+    if (FAILED(D3DCompileFromFile(
+        L"Assets/Shaders/Main.ps.hlsl",
+        nullptr,
+        D3D_COMPILE_STANDARD_FILE_INCLUDE,
```

Comment:
Ah include handlers... I think it makes sense to cover them at some point in time as well, further down the road :) as an appendix or so